tests/west/project/test_project.py

```diff
@@ -93,17 +82,28 @@ def clean_west_topdir(tmpdir):
         if path == 'Kconfiglib':
             subprocess.check_call([git, 'branch', 'zephyr'], cwd=fullpath)
 
-    # Create the per-tmpdir manifest file.
-    with open(MANIFEST_TEMPLATE_PATH, 'r') as src:
-        with open(str(tmpdir.join('manifest.yml')), 'w') as dst:
-            dst.write(src.read().format(tmpdir=str(repos)))
-
-    # Initialize and change to the installation directory.
-    zephyrproject = tmpdir.join('zephyrproject')
-    zephyrproject.mkdir()
-    zephyrproject.mkdir('west')
-    zephyrproject.join('west', '.west_topdir').ensure()
-    zephyrproject.chdir()
+    # Create west/.west_topdir, to mark this directory as a West installation,
+    # and a manifest.yml pointing to the repositories we created above
+    tmpdir.mkdir('west').join('.west_topdir').ensure()
+    tmpdir.join('manifest.yml').write('''
+manifest:
+  defaults:
+    remote: repos
+    revision: master
+
+  remotes:
+    - name: repos
+      url: file://{}
+
+  projects:
+    - name: net-tools
+    - name: Kconfiglib
+      revision: zephyr
+      path: sub/Kconfiglib
+'''.format(remote_repos_dir))
+
+    # Switch to the top-level West installation directory
```

Comment:
Do you think they would ever overlap in practice? This code shorter, simpler to understand, and not any less robust, and that's what counts.

Too much namespacing and directories is a bad thing I think. I think people get carried away with it sometimes, in some quest to look like a serious developer ("bro r u even namespacing properly?"). :)

If there's negligible risk of name collisions, don't namespace. As a bonus, you get tests for whether the code gets confused by non-projects being in there. ;)

Reply:
Heck, in this case the risk is precisely zero, because we control what projects get put in there for the tests.

Reply:
> Heck, in this case the risk is precisely zero, because we control what projects get put in there for the tests.

... and we have to remember that the tests behave differently than a fresh environment would.

I don't agree that it's simpler to understand. I could see a reader thinking that we were manually placing repositories as if they had been cloned by west.

Reply:
They behave the same.

Here's the old layout:

```
repositories/kconfiglib/
repositories/net-tools/
repositories/manifest.yml
zephyrproject/west/.west_topdir
zephyrproject/west/west/
zephyrproject/west/manifest/
```

Here's the new layout:

```
repos/kconfiglib
repos/net-tools
manifest.yml
west/.west_topdir
west/west/
west/manifest/
```

The only way they could act different would be if we accidentally added a test that reuses the `repos` name for something else, which is (1) very unlikely, and (2) trivial to fix.

Namespacing when you don't need it is bad. Don't namespace unless you have a huge project where you could realistically get naming collisions.

Heck, I could rename `repos` to `upstream-repos`. That way it'll be obvious even if you don't read the comment in `clean_west_topdir()`, and it drops the risk of collisions somewhere into the national lottery range:

```
upstream-repos/kconfiglib
upstream-repos/net-tools
manifest.yml
west/.west_topdir
west/west/
west/manifest/
```

One thing that confuses me for real is that West/Zephyr mixup by the way. `zephyrproject` made me think it was the Zephyr repository when I first saw it.

Reply:
There, it's now:

```
remote-repos/kconfiglib
remote-repos/net-tools
remote-repos/west
remote-repos/manifest
manifest.yml
west/.west_topdir
west/west/
west/manifest/
```